File: bmlsj/01/[PGS - lv2] 01_디펜스 게임.java
import java.util.*;
class Solution {
    public int solution(int n, int k, int[] enemy) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for(int i = 0; i < enemy.length; i++) {
            pq.offer(enemy[i]);
            
            if (pq.size() > k) {
                n -= pq.poll();
                
                if (n < 0) {
                    return i;
                }
            }
        }
        
        
        return enemy.length;
    }
}

File: bmlsj/01/[PGS - lv2] 01_디펜스 게임.py
import heapq
def solution(n, k, enemy):
    answer = 0
    
    pq = []
    for i, e in enumerate(enemy):
        heapq.heappush(pq, e)
        if len(pq) > k:
            n -= heapq.heappop(pq)
            
            if n < 0:
                return i
        
        
    return len(enemy)